aztec_backend_wasm/src/lib.rs

```diff
@@ -1,14 +1,35 @@
 #![warn(unused_crate_dependencies, unused_extern_crates)]
 #![warn(unreachable_pub)]
 
-use common::acvm::acir::circuit::Circuit;
+use common::acvm::acir::{circuit::Circuit, native_types::Witness};
+use common::proof::flatten_witness_map;
 use common::serializer::serialize_circuit;
 use wasm_bindgen::prelude::*;
 
 #[wasm_bindgen]
 pub fn serialize_acir_to_barretenberg_circuit(acir_bytes: Vec<u8>) -> Vec<u8> {
     console_error_panic_hook::set_once();
 
-    let circuit = Circuit::read(&*acir_bytes).unwrap();
+    let circuit = Circuit::read(&*acir_bytes).expect("could not deserialize circuit");
     serialize_circuit(&circuit).to_bytes()
 }
+
+/// Returns the uncompressed flattened witness format expected by Barretenberg.
+///
+/// This function is necessary as witness maps are saved to disk in a compressed format;
+/// Barretenberg doesn't expect a compressed witness map however so it must be converted.
```

Comment:
@TomAFrench Should this "decompression" be the responsibility of *each* backend or noir/nargo? It seems like most backends won't want to deal with a compressed format.

Reply:
Yes, this was something I was thinking about as well. Decompression should be generic (as we're just undoing some processing done by `nargo` however we can't be sure that all backends will want the witness flattened in this way.

I think that while `ProofSystemCompiler::prove_with_pk` accepts the witness as a `BTreeMap<Witness, FieldElement>` then we should treat this as the responsibility of the backend. At least while when proving/verifying in the browser we're using a BB-specific npm package, we could probably handle this more nicely once we move to Joss's package.

Reply:
We have precedent for this being here as this function is basically equivalent to `packed_witness_to_witness` which was removed a little while back.

Reply:
The compile function returns the noir program's circuit rather than its witness so it's related to the current goal where we're wanting to prove/verify against an existing circuit.

While we wouldn't have to do this work if we were proving against a witness which has been generated within javascript, that adds a lot of extra work (abi encoding params, executing the circuit, etc.) so it's easier to test using pregenerated witnesses.